tests/test_workspace.py

```
@@ -0,0 +1,484 @@
+    def test_resolve_source_path(self):
+        with TemporaryDirectory() as tmpdir:
+            ws_file = Path(tmpdir) / ".cypilot-workspace.toml"
+            cfg = WorkspaceConfig(
+                sources={"repo": SourceEntry(name="repo", path="sub/repo")},
+                workspace_file=ws_file,
+            )
+            resolved = cfg.resolve_source_path("repo")
+            assert resolved == (Path(tmpdir) / "sub" / "repo").resolve()
+
+    def test_resolve_source_path_unknown(self):
+        cfg = WorkspaceConfig(workspace_file=Path("/tmp/ws.toml"))
```

Ruff (0.15.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/ws.toml"

(S108)